includes/admin/class-wc-payments-bnpl-announcement.php

```diff
@@ -0,0 +1,238 @@
+				'wc-payments-deposits',
+				'wc-payments-transactions',
+				'wc-payments-deposit-details',
+				'wc-payments-transaction-details',
+				'wc-payments-multi-currency-setup',
+			],
+			true
+		) ) ) {
+			return;
+		}
+
+		// at least 3 purchases (on any payment method).
+		$woopayments_successful_orders_count = $this->get_woopayments_successful_orders_count();
+		if ( $woopayments_successful_orders_count < 3 ) {
+			return;
+		}
+
+		// don't display the promo if the merchant already has BNPL methods enabled.
+		$enabled_bnpl_payment_methods = array_intersect(
+			\WCPay\Constants\Payment_Method::BNPL_PAYMENT_METHODS,
+			$this->gateway->get_upe_enabled_payment_method_ids()
+		);
+		if ( ! empty( $enabled_bnpl_payment_methods ) ) {
+			return;
+		}
+
+		add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_scripts' ] );
+
+		add_user_meta( get_current_user_id(), '_wcpay_bnpl_april15_viewed', '1' );
+	}
+
+	/**
+	 * Enqueues the script & styles for the BNPL announcement dialog.
+	 *
+	 * @return void
+	 */
+	public function enqueue_scripts() {
+		WC_Payments::register_script_with_dependencies( 'WCPAY_BNPL_ANNOUNCEMENT', 'dist/bnpl-announcement' );
+		wp_set_script_translations( 'WCPAY_BNPL_ANNOUNCEMENT', 'woocommerce-payments' );
+		WC_Payments_Utils::register_style(
+			'WCPAY_BNPL_ANNOUNCEMENT',
+			plugins_url( 'dist/bnpl-announcement.css', WCPAY_PLUGIN_FILE ),
+			[ 'wc-components' ],
+			WC_Payments::get_file_version( 'dist/bnpl-announcement.css' ),
+			'all'
+		);
+		// conditionally show afterpay/clearpay based on account country.
+		wp_localize_script(
+			'WCPAY_BNPL_ANNOUNCEMENT',
+			'wcpayBnplAnnouncement',
+			[
+				'accountCountry' => $this->account->get_account_country(),
+			]
+		);
+
+		wp_enqueue_script( 'WCPAY_BNPL_ANNOUNCEMENT' );
+		wp_enqueue_style( 'WCPAY_BNPL_ANNOUNCEMENT' );
+	}
+
+	/**
+	 * Returns the number of successful orders paid with any WooPayments payment method.
+	 *
+	 * @return int
+	 */
+	private function get_woopayments_successful_orders_count() {
+		// using a transient to store the value of a previous calculation, since it can be expensive on each page load.
+		$wcpay_successful_orders_count = get_transient( 'wcpay_bnpl_april15_successful_purchases_count' );
+		if ( false !== $wcpay_successful_orders_count ) {
+			return intval( $wcpay_successful_orders_count );
+		}
+
+		$is_sandbox_mode = false;
+		try {
+			$is_sandbox_mode = WC_Payments::mode()->is_dev();
+		} catch ( \Exception $e ) {
+			// nothing to do here, we'll just assume it's live mode.
+			$is_sandbox_mode = false;
+		}
+
+		$orders = wc_get_orders(
+			[
+				// we don't need them all, just more than 3.
+				'limit'          => 5,
+				'status'         => [ 'completed', 'processing' ],
+				// phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+				'meta_query'     => [
+					[
+						'key'     => '_wcpay_mode',
+						'value'   => $is_sandbox_mode ? 'test' : 'prod',
+						'compare' => '=',
+					],
+				],
```

Comment:
Mm, re-reading the acceptance criteria, I think I am going to remove these queries entirely.

> Targets only existing merchants with WooPayments plugin installed and activated, live in production and that have at least 3 purchases (**on any gateway**).

I don't think we need to be limiting by orders made by the WooPayments gateway. I suppose this makes sense, since the entire purpose of this is to drive users to start using our gateway, so shouldn't necessarily limit this notice to those who are already using WooPayments. 🤷

Reply:
Updated in 3338b455f9de3e39fb6ce2633891b3360611cacb.